Bug Fix and Enhancement to code base

ashokmor opened the pull request:
Updated to azbs  to align with what it stand for "Azure Blob Storage"


Important change is Override getChildren() method for AzFileObject to throw FileNotFolderException for File Blob Object. This one is real problem when we make getChildren() call on File and it returns empty, it's difficult to identify whether it's real file object or an empty folder in VFS.

Commit:
Updated schema from azsb to azbs and upggraded pom version

Commit:
Updated readme for azsb to azbs change

Commit:
Override getChildren() method for AzFileObject to throw FileNotFolderException for File Blob Object

Commit:
Added temp solution to get Folder as type for varitual path

Commit:
Rolled back changes in doCreateFolder method

ashokmor committed:
Upgraded azure storage dependency
Provided facility to override default values of various block upload related properties
Override Copy method in AzFileObject to improve upload performance

ashokmor committed:
Added support to detected content type from file name

ashokmor committed:
Tess case to detect content type from file name

Commit:
- set default UPLOAD_BLOCK_SIZE to 2 MB to set in (currBlob.setStreamWriteSizeInBytes)
- Added "azure.concurrent.request.count" and "azure.enable.logging" new properties with constant. Default values set to 5 and false respectively

Commit:
Rolled back azure-storage version changes

Commit:
Updated parallel upload thread and upload block size default value

Commit:
- Updated azure-storage library version to latest release
- Updated this library version to 1.3.1
- Set default value for parallel upload thread
- Removed "azure.concurrent.request.count" property usage from AzFileObject, it is being already defined in AzFileSystem as "azure.parallel.upload.thread"

ashokmor committed:
Added check to return file type as a folder when path ends with "/"

Commit:
Fixed copy directories

Commit:
Added javadoc.

ashokmor committed:
Merge pull request #1 from rogeriopt/master

Fixed copy directories

ashokmor committed:
Updating POM version to release 1.3.3
Removed unused code

Commit:
Fix not setting content-type

Commit:
Fix not set content-type

ashokmor committed:
Merge pull request #2 from rogeriopt/master

Fix not set content-type

ashokmor committed:
Updated POM version for release of minor fix

ashokmor commented:
@ptahchiev 

Can you please review PR?

ptahchiev commented:
My repository is a fork of this one: https://github.com/kervinpierre/vfs-azure
I believe your PR should be against that repository

ashokmor commented:
@ptahchiev actually I forked from you so for my PR it should be your repo. They might not have the changes which I got from yours.

ashokmor commented:
@ptahchiev 

Can you please review this PR and merge.

PR to your forked one should go from you from your forked branch.

Commit:
Updated the call to initiate cloud storage account to support file operation for https enabled storage account
Updated minor version in pom file to support the change in new version

ashokmor commented:
@ptahchiev 
Can you please review my PR (all commits) and approve it?

I have raised PR to original repo - https://github.com/kervinpierre/vfs-azure/pull/3

kervinpierre commented:
Hi, sorry I hadn't checked this repo in a while.

If you have a list of PRs you'd like to merge I'd be happy to get them all at once to merge downstream.

ashokmor commented:
Thanks @kervinpierre,

Right now only this PR. Could you please merge.

Commit:
Added functionality to quick copy file within same account and container
Added method to check server to server copy is possible
Override canRenameTo method to use copyFrom for move operation
Updated pom version to release new change in patch

ashokmor committed:
Merge pull request #3 from RajuTanchak/master

Added functionality to quick copy file within same account and container

Commit:
Added functionality to quick copy file within same account

ashokmor committed:
Merge pull request #4 from RajuTanchak/master

Added functionality to quick copy file within same account

Commit:
No need to check file system to check for server to server copy

ashokmor commented:
@ptahchiev 

Added server side copy, Copying file from Azure to Azure would get benefited as they no longer need to streams and read/write datas.?

Commit:
Added support to generate signed url

oliverlockwood committed:
Note that this fork is deprecated in favour of dalet-oss/vfs-azure

ashokmor committed:
Merge pull request #5 from dalet-oss/deprecate-fork

Note that this fork is deprecated in favour of dalet-oss/vfs-azure

kervinpierre commented:
If anyone cares to you can take over my branch, I don't have to time right now unfortunately

oliverlockwood commented:
@kervinpierre: we'd be happy to take over the repo.  (I work with @ashokmor.)

As per my comment on https://github.com/kervinpierre/vfs-azure/issues/4 we've already gone down that road.  If you'd like to transfer your repo to either @ashokmor, myself or (preferably) directly to the @dalet-oss organisation, then we can keep it as the origin of other people's forks etc, which would help the community.

kervinpierre commented:
I invited @ashokmor , I couldn't add/find the dalet-oss user
